validator/tests/test.bal

```diff
@@ -216,3 +216,137 @@ function testTargetType() {
+
+    test:assertTrue(parsedError == ["Missing required Element: 'resourceType'"]);
+}
+
+@test:Config{}
+function testMissingElementParsing() {
+    string errorMessage = string `Failed to parse request body as JSON resource due to 'map<json>' value cannot be converted to 'health.fhir.r4.international401:Patient':
+                missing required field 'text.status' of type 'ballerinax/health.fhir.r4:4:StatusCode' in record 'health.fhir.r4:Narrative'`;
+    string[] parsedError = processFHIRParserErrors(errorMessage);
+
+    test:assertTrue(parsedError == ["Missing required field 'text.status'"]);
+}
+
+@test:Config{}
+function testInvalidFieldParsing() {
+    string errorMessage = string `Failed to parse request body as JSON resource due to 'map<json>' value cannot be converted to 'health.fhir.r4.international401:Patient':
+                value of field 'isd' adding to the record 'health.fhir.r4.international401:Patient' should be of type 'health.fhir.r4:Element', found '"example"'`;
+    string[] parsedError = processFHIRParserErrors(errorMessage);
+
+    test:assertTrue(parsedError == ["Invalid field 'isd'. Type of field should be 'health.fhir.r4:Element'"]);
+}
+
+@test:Config{}
+function testInvalidFieldValueParsing() {
+    string errorMessage = string `Failed to parse request body as JSON resource due to 'map<json>' value cannot be converted to 'health.fhir.r4.international401:Patient':
+                field 'telecom[1].system' in record 'health.fhir.r4:ContactPoint' should be of type 'ballerinax/health.fhir.r4:4:ContactPointSystem', found '1'`;
+    string[] parsedError = processFHIRParserErrors(errorMessage);
+
+    test:assertTrue(parsedError == ["Invalid value of field 'telecom[1].system'. Type of value should be 'ballerinax/health.fhir.r4:4:ContactPointSystem'"]);
+}
+
+@test:Config{}
+function testInvalidArrayElements() {
+    string errorMessage = string `Failed to parse request body as JSON resource due to 'map<json>' value cannot be converted to 'health.fhir.r4.international401:Patient':
+                array element 'name[0].given[0]' should be of type 'string', found '1'`;
+    string[] parsedError = processFHIRParserErrors(errorMessage);
+
+    test:assertTrue(parsedError == ["Invalid array element 'name[0].given[0]'. Type of element should be 'string'"]);
+}
+
+@test:Config{}
+function testMultityypeScenarioFieldParsing() {
+    string errorMessage = string `Failed to parse request body as JSON resource due to 'map<json>' value cannot be converted to 'health.fhir.r4.international401:Patient':
+                {
+                  missing required field 'contact[0].name._family.extension[0].valueCodeableConcept' of type 'health.fhir.r4:CodeableConcept' in record 'health.fhir.r4:CodeableConceptExtension'     
+                  value of field 'valuaeString' adding to the record 'health.fhir.r4:CodeableConceptExtension' should be of type 'health.fhir.r4:Element', found '"VV"'
```

Comment:
```suggestion
                  value of field 'valueString' adding to the record 'health.fhir.r4:CodeableConceptExtension' should be of type 'health.fhir.r4:Element', found '"VV"'
```

Fix other places in a seperate PR

Reply:
This is a sample error message, it is intentional.